snack-lib/default.nix

```diff
@@ -14,43 +8,15 @@
 , symlinkJoin
 , writeScript
 , runCommand
+, callPackage
 }:
-let
-  # Takes a (string) filepath and creates a derivation for that file (and for
-  # that file only)
-  singleOut = base: file:
-    let
-      basePrefix = (builtins.toString base) + "/";
-      pred = file: path: type:
-        let
-          actual = (lib.strings.removePrefix basePrefix path);
-          expected = file;
-        in
-          (expected == actual) ||
-          (type == "directory" && (lib.strings.hasPrefix actual expected));
-      mod = fileToModule file;
-      # TODO: even though we're doing a lot of cleaning, there's sitll some
-      # 'does-file-exist' happening
-      src0 = lib.cleanSource base;
 
-    in stdenv.mkDerivation {
-      name = mod;
-      src = lib.cleanSourceWith  { filter = (pred file); src = src0; };
-      builder = writeScript (mod + "-builder")
-      # TODO: make sure the file actually exists and that there's only one
-      ''
-        echo "Singling out module ${mod} (file is ${file})"
-        source $stdenv/setup
-        mkdir -p $out
-        echo "Running: cp $src/${file} $out/${file}"
-        echo "Listing $src"
-        ls $src/**/*
-        mkdir -p $(dirname $out/${file})
-        cp $src/${file} $out/${file}
-        echo "Done: Singling out module ${mod} (file is ${file})"
-      '';
-    };
+with (callPackage ./files.nix {});
 
+# why is "inherit" needed?
```

Comment:
@zimbatm any idea why I need to do this?

Reply:
`callPackage` fills the argument of the target file using the reflection API from it's scope, which is `pkgs` in that case.